File: Carros Objective-C/Carros Objective-C/Controller/DetalhesCarroViewController.m
//
//  DetalhesCarroViewController.m
//  Carros Objective-C
//
//  Created by Antonio Lopes on 8/5/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

#import "DetalhesCarroViewController.h"
#import "ListaCarrosViewController.h"
#import "MapViewController.h"
#import "GpsMapViewController.h"
#import "VideoViewController.h"
#import "VideoUtil.h"
#import "Utils.h"
#import "PopoverUtil.h"

@interface DetalhesCarroViewController ()

@end

@implementation DetalhesCarroViewController

@synthesize carro;

- (void)viewDidLoad {
    [super viewDidLoad];
    
    if (carro != nil) {
        [self updateCarro:carro];
    }
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
    if ([Utils isIphone]) {
        
    }
}

-(IBAction)visualizarMapa:(id)sender {
//    MapViewController *vc = [[MapViewController alloc] initWithNibName:@"MapViewController" bundle:nil];
    GpsMapViewController *vc = [[GpsMapViewController alloc] initWithNibName:@"MapViewController" bundle:nil];
    [vc setCarro:[self carro]];
    
    if ([Utils isIphone]) {
        [self.navigationController pushViewController:vc animated:true];
    } else {
        [PopoverUtil show:self andViewController:vc andSourceView:(UIView*)sender andWidth:600 andHeight:500];
    }
}

-(IBAction)visualizarVideo:(id)sender {
    VideoUtil *videoUtil = [[VideoUtil alloc] init];
    NSURL *url = [[NSURL alloc] initWithString:[self.carro url_video]];
    [videoUtil playUrlFullScreen:url withViewController:self];
//    VideoViewController *vc = [[VideoViewController alloc] initWithNibName:@"VideoViewController" bundle:nil];
//    [vc setCarro:[self carro]];
//    [self.navigationController pushViewController:vc animated:true];
}

-(void)updateCarro:(Carro*)selectedCarro {
    [self setCarro:selectedCarro];
    [self setTitle:selectedCarro.nome];
    [descricao setText:selectedCarro.desc];
    [image setUrl:selectedCarro.url_foto];
}

-(void)splitViewController:(UISplitViewController *)svc willChangeToDisplayMode:(UISplitViewControllerDisplayMode)displayMode {
    if (displayMode == UISplitViewControllerDisplayModePrimaryHidden) {
        UIBarButtonItem *btPopOver = [[UIBarButtonItem alloc] initWithTitle:@"Lista" style:UIBarButtonItemStylePlain target:self action:@selector(onClickPopover)];
        [self.navigationItem setLeftBarButtonItem:btPopOver];
    } else {
        [self.navigationItem setLeftBarButtonItem:nil];
    }
}

-(void)onClickPopover {
    ListaCarrosViewController *listaViewController = [[ListaCarrosViewController alloc] init];
    [PopoverUtil show:self andViewController:listaViewController andSource:self.navigationItem.leftBarButtonItem andWidth:400 andHeight:800];
}
@end

File: Carros Swift/Carros Swift/Controller/DetalhesCarroViewController.swift
//
//  DetalhesCarroViewController.swift
//  Carros Swift
//
//  Created by Antonio Lopes on 8/5/16.
//  Copyright © 2016 Antonio Lopes. All rights reserved.
//

import UIKit

class DetalhesCarroViewController: UIViewController {

    // MARK: Outlets
    @IBOutlet var image : DownloadImageView!
    @IBOutlet var descricao : UITextView!
    
    // MARK: Variables
    var carro : Carro?
    
    override func viewDidLoad() {
        super.viewDidLoad()

        if let selectedCarro = carro {
            self.title = selectedCarro.nome
            self.image.setUrl(selectedCarro.url_foto!)
            self.descricao.text = selectedCarro.desc
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    @IBAction func visualizarMapa() {
        let vc = MapViewController(nibName: "MapViewController", bundle: nil)
        vc.carro = self.carro
        self.navigationController!.pushViewController(vc, animated: true)
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
